llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -5350,6 +5350,12 @@ static bool isLocalRepeatingShuffle(ArrayRef<int> Mask, int Span) {
   return true;
 }
 
+/// Is this mask only using elements from the first span of the input?
+static bool isLowSourceShuffle(ArrayRef<int> Mask, int Span) {
+  return !any_of(Mask,
```

Comment:
Use `none_of` instead of `!any_of` or invert the lambda and use all_of.

```
  return all_of(Mask,
                [&](const auto &Idx) { return Idx == -1 || Idx < Span; });
```